[Feat] 체크박스 컴포넌트 제작

BomEllen opened the pull request:
## PR 유형
- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## PR 체크리스트
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

### PR 상세
- 체크박스 컴포넌트를 제작하였습니다.
- props로 값을 전달하여 상위 컴포넌트에서 사용할 수 있습니다.


## 이슈
resolved #46

BomEllen committed:
[Feat] 체크박스 컴포넌트 제작

- 체크박스 컴포넌트를 제작하였습니다.
- props로 값을 전달하여 상위 컴포넌트에서 사용할 수 있습니다.

resolved #46

BomEllen committed:
[Refactor] svg 렌더링 최적화

- svg 요소가 checked 상태에 따라 조건부 렌더링 되던 코드를 svg는 항상 렌더링 되면서 checked 상태에 따라 className으로 보이거나 숨기는 방식으로 수정 하므로써 불필요한 렌더링을 줄임

resolved #46

Sungwoo00 reviewed (changes requested):
label과 input을 htmlFor랑 id로 연결해 주셨으면 좋겠습니다~

BomEllen committed:
[Refactor] id로 label과 input 명시적 연결

- useId() 훅을 사용하여 컴포넌트를 활용할 때 고유한 id를 생성하도록 수정하였습니다.
- label과 input을 htmlFor와 id로 명시적으로 연결하여 접근성을 향상시켰습니다.

resolved #46

Sungwoo00 reviewed (approved):
나이스샷~ 저도 useId로 바꿔 놨어요~

BomEllen committed:
[Refactor] svg 이미지 assets에 추가 하여 불러옴

- svg를 웹에서 불러오는것이 아닌 assets에 추가하여 img 태그로 불러왔습니다.

resolved #46

BomEllen committed:
[Comment] 주석 삭제

- 필요 없는 주석을 삭제하였습니다.

resolved #46

MinQyu reviewed (approved):
고생하셨어요~
